### ISSUE TITLE
False positive with 2 (or more) foreach with sep macro

### ISSUE THREAD

**Issue by jakubvojacek** [efabrica-team/phpstan-latte#104]
Hello 

I am getting the following error when foreach inside foreach is used together with separator macro. Perhaps it should be added to ignored errors?

```
 ------ ------------------------------------------------------------------------------------- 
  Line   AppManager.latte rendered from Motv\FrontModule\Components\Tools\AppManager::render  
 ------ ------------------------------------------------------------------------------------- 
  3      Cannot call method getParent() on Latte\Essential\CachingIterator|null.              
 ------ ------------------------------------------------------------------------------------- 
```

```
{foreach $getAppsByVendors as $vendorName => $appManager}
    {foreach $appManager as $appArray}
        {sep} | {/sep}
    {/foreach}
{/foreach}
```

```php
<?php

declare (strict_types=1);
use Latte\Runtime as LR;
/** 
* Motv\FrontModule\Components\Tools\AppManager::render
* @property PHPStanLatteTemplate_cf9b344f8e2e16f01d8dd1bfb0076823_global $global
*/
final class PHPStanLatteTemplate_cf9b344f8e2e16f01d8dd1bfb0076823 extends Latte\Runtime\Template
{
    public function main(array $ʟ_args) : void
    {
        extract($this->params);
        /** @var string $baseUrl */
        /** @var string $basePath */
        /** @var Latte\Runtime\FilterInfo $ʟ_fi */
        /** @var array<string> $ʟ_tag */
        /** @var array $ʟ_if */
        /** @var array $ʟ_ifc */
        /** @var array $ʟ_try */
        /** @var array $ʟ_loc */
        /** @var Nette\Security\User $user */
        /** @var Nette\Application\UI\Presenter $presenter */
        /** @var Nette\Application\UI\Control $control */
        /** @var array<stdClass> $flashes */
        /** @var Nette\Localization\Translator $netteLocalizationTranslatorFilter */
        /** @var Motv\FrontModule\Components\Tools\AppManager $presenter */
        /** @var Motv\FrontModule\Components\Tools\AppManager $control */
        /** @var array<string, array<array<string, array<Motv\ApiModule\Entity\FileManagerVersionEntity>|Motv\ApiModule\Entity\AppManagerEntity>>> $getAppsByVendors */
        unset($ʟ_args);
        if ($this->global->snippetDriver?->renderSnippets($this->blocks[self::LayerSnippet], $this->params)) {
            return;
        }
        /* line 1 */
        foreach ($iterator = $ʟ_it = new Latte\Essential\CachingIterator($getAppsByVendors, $ʟ_it ?? null) as $vendorName => $appManager) {
            /* line 2 */
            foreach ($iterator = $ʟ_it = new Latte\Essential\CachingIterator($appManager, $ʟ_it ?? null) as $appArray) {
                echo '        ';
                /* line 3 */
                if (!$iterator->isLast()) {
                    echo ' | ';
                }
                echo "\n";
            }
            $iterator = $ʟ_it = $ʟ_it->getParent();
        }
        $iterator = $ʟ_it = $ʟ_it->getParent();
    }
    public function prepare() : array
    {
        extract($this->params);
        /** @var string $baseUrl */
        /** @var string $basePath */
        /** @var Latte\Runtime\FilterInfo $ʟ_fi */
        /** @var array<string> $ʟ_tag */
        /** @var array $ʟ_if */
        /** @var array $ʟ_ifc */
        /** @var array $ʟ_try */
        /** @var array $ʟ_loc */
        /** @var Nette\Security\User $user */
        /** @var Nette\Application\UI\Presenter $presenter */
        /** @var Nette\Application\UI\Control $control */
        /** @var array<stdClass> $flashes */
        /** @var Nette\Localization\Translator $netteLocalizationTranslatorFilter */
        /** @var Motv\FrontModule\Components\Tools\AppManager $presenter */
        /** @var Motv\FrontModule\Components\Tools\AppManager $control */
        /** @var array<string, array<array<string, array<Motv\ApiModule\Entity\FileManagerVersionEntity>|Motv\ApiModule\Entity\AppManagerEntity>>> $getAppsByVendors */
        if ($this->getReferringTemplate() === null || $this->getReferenceType() === 'extends') {
            foreach (array_intersect_key(['vendorName' => '1', 'appManager' => '1', 'appArray' => '2'], $this->params) as $ʟ_v => $ʟ_l) {
                trigger_error("Variable \${$ʟ_v} overwritten in foreach on line {$ʟ_l}");
            }
        }
        return get_defined_vars();
    }
}
/**
* @property-read array $coreParentFinder
* @property-read Motv\FrontModule\Components\Tools\AppManager $uiControl
* @property-read Motv\FrontModule\Components\Tools\AppManager $uiPresenter
* @property-read Nette\Bridges\ApplicationLatte\SnippetDriver|null $snippetDriver
* @property-read string|null $uiNonce
* @property-read Kdyby\Redis\RedisStorage $cacheStorage
*/
class PHPStanLatteTemplate_cf9b344f8e2e16f01d8dd1bfb0076823_global extends \stdClass
{
}
```

**Comment by lulco** [comment]
Thanks for report, probably yes, we can ignore it. What do you think @MartinMystikJonas

**Comment by MartinMystikJonas** [comment]
Yes ignore.
